engine/src/main/java/org/terasology/rendering/logic/SkeletonRenderer.java

```diff
@@ -248,7 +248,7 @@ public void renderOpaque() {
             skeletalMesh.material.setFloat("blockLight", 1.0f, true);
             skeletalMesh.material.setFloat3("colorOffset", skeletalMesh.color.rf(), skeletalMesh.color.gf(), skeletalMesh.color.bf(), true);
 
-            skeletalMesh.material.setMatrix4("projectionMatrix", worldRenderer.getActiveCamera().getProjectionMatrix());
+            skeletalMesh.material.setMatrix4("projectionMatrix", new org.joml.Matrix4f(worldRenderer.getActiveCamera().getProjectionMatrix()).transpose());
```

Comment:
Since this is in the rendering path, the `new` caught my eye and I started wondering if there's a way to do this without that heap allocation.

If I'm reading this correctly, the projection matrix doesn't change that often, right? And even if it _did_ change every frame, we should plan for rendering a lot more than one skeleton per frame.

This idea sounds a little sketchy but maybe it's worth something: add a transposedProjectionMatrix accessor to the camera. that way these things that need the transposed-version can get it without each doing their own transposition of it.

Reply:
there is another transpose in the perspective camera that needs to be undone. 

```
return new Matrix4f().perspective(fovY,aspectRatio,zNear,zFar).transpose();
```